### ISSUE TITLE
static library problems

### ISSUE THREAD

**Issue by pavlis** [wangyinz/mspass#170]
Our build procedure to build static libraries that are installed with a `make install` has a minor organizational problem that should be fixed.   The best way to see the issue is a couple of outputs.  After `cd $MSPASS_HOME/lib` one sees this standard install:
```
$ ls
libalgorithms.a  libmspass.a  libseismic.a  libutility.a
```
The problem is libalgorithms that has this content:
```
$ ar -t libalgorithms.a 
dummy.o
```
Things in our source tree grouped in algoirthms directories seem to have been placed in libmspass.a as shown by this output:
```
$ ar -t libmspass.a 
BasicTimeSeries.o
CoreSeismogram.o
CoreTimeSeries.o
PowerSpectrum.o
Seismogram.o
SlownessVector.o
Taper.o
TimeSeries.o
AntelopePf.o
AttributeCrossReference.o
AttributeMap.o
ErrorLogger.o
Metadata.o
MetadataDefinitions.o
MsPASSError.o
ProcessManager.o
ProcessingHistory.o
SphericalCoordinate.o
data_directory.o
dmatrix.o
metadata_helpers.o
normalize.o
Butterworth.o
agc.o
bundle.o
seismogram_helpers.o
slice_and_dice.o
sparse_convolve.o
tseries_helpers.o
amplitudes.o
CNR3CDecon.o
ComplexArray.o
ComputeTaperLength.o
FFTDeconOperator.o
GeneralIterDecon.o
LeastSquareDecon.o
MTPowerSpectrumEngine.o
MultiTaperSpecDivDecon.o
MultiTaperXcorDecon.o
ScalarDecon.o
ShapingWavelet.o
ThreeCSpike.o
WaterLevelDecon.o
circular_shift.o
dpss.o
nextPowerOf2.c.o
wavelet.o
```
That isn't a functional problem for mspass directly since the python setup.py script ignores all that and build .so files that are the python ccore modules anyway.   It came up with me as I encountered linking problems when I tried to link to libalgorithms.  Since it has only "dummy" it is useless.  The solution was easy (just told the loader to use libmspass), but this is a bit of an organizational confusion of the library layout.

**Comment by wangyinz** [comment]
Well, it seems your local branch is out of sync with the remote. We have exchanged this issue over emails previously, and I have already changed the build structure in https://github.com/wangyinz/mspass/commit/2199590721cda4836b1523171143f3c02dc5b8ba. There should be only one `libmspass.a` created in this new build. The other libraries of generic names are no longer created during the build and would not be installed neither.
